### ISSUE TITLE
Should Math.mod only return positive numbers?

### ISSUE THREAD

**Issue by NigelBreslaw** [slint-ui/slint#6178]
There seems to be quite some difference between languages on if modulo returns the remainder or the signed remainder. Reporting this after the comments to https://github.com/slint-ui/slint/pull/6166#discussion_r1761418599

**Comment by ogoffart** [comment]
I think it make sense to change this behavior.
I can't think of a good use case of the mod function.

Mathematically, the "mod" is the remainder of the integer division.
And so if the division truncates, it should indeed be negative to be able to do something like:
```slint
property <int> div: a/b;
property <bool> assert: (div * b + a.mod(b)) == a;
```
So that assert is true for every value.
https://en.wikipedia.org/wiki/Modulo#Variants_of_the_definition

But who cares about that anyway? In UI code, I just want to use mod to limit my angle in a given interval, or any length or value, really.

So in conclusion, I think we should change the definition of our mod function to always return a positive value.
